bluemira/equilibria/grad_shafranov.py

```diff
@@ -93,33 +92,36 @@ def __call__(self, nx, nz):
             nz = np.floor_divide(nz + 1, 2)
 
         A = lil_matrix((nx * nz, nx * nz))
-        A.setdiag(ones(nx * nz))
+        A.setdiag(1)
         # NOTE: nb.jit doesn't seem to help here :(
-        for i in range(1, nx - 1):
-            for j in range(1, nz - 1):
-                ind = i * nz + j
-                rp = 0.5 * (x[i + 1] + x[i])  # x_{i+1/2}
-                rm = 0.5 * (x[i] + x[i - 1])  # x_{i-1/2}
-                A[ind, ind] = -(x[i] / d_x2) * (1 / rp + 1 / rm) - 2 / d_z2  # j, l
-                A[ind, ind + nz] = (x[i] / d_x2) / rp  # j, l-1
-                A[ind, ind - nz] = (x[i] / d_x2) / rm  # j, l+1
-                A[ind, ind + 1] = 1 / d_z2  # j-1, l
-                A[ind, ind - 1] = 1 / d_z2  # j+1, l
-            # Apply symmetry boundary if desired
-            if self.force_symmetry:
-                # Apply ghost point method to apply symmetry to (d/dz^2) operator
-                # constributions close to symmetry plane.
-                # If symmetry boundary is centred halfway between cells,
-                # d(psi)/dz = 0 across midplane,
-                # else if symmetry boundary is centred on cells, d(psi)/dz
-                # is equal either side of midplane but reverses sign.
-                ind = i * nz + nz - 1
-                ghost_factor = 1 + nz % 2
-
-                A[ind, ind] = -(x[i] / d_x2) * (1 / rp + 1 / rm) - ghost_factor / d_z2
-                A[ind, ind - 1] = ghost_factor / d_z2
-                A[ind, ind + nz] = (x[i] / d_x2) / rp  # j, l-1
-                A[ind, ind - nz] = (x[i] / d_x2) / rm  # j, l+1
+        i, j = np.meshgrid(np.arange(1, nx - 1), np.arange(1, nz - 1), indexing="ij")
+        i = i.ravel()
+        j = j.ravel()
+        ind = i * nz + j
+        _dx_2_m_halfx = _dx_2 - half_xdx[i]
+        _dx_2_p_halfx = _dx_2 + half_xdx[i]
+        A[ind, ind] = -2 * (_dx_2 + _dz_2)  # j, l
+        A[ind, ind + nz] = _dx_2_m_halfx  # j, l-1
+        A[ind, ind - nz] = _dx_2_p_halfx  # j, l+1
```

Comment:
Doesn't seem consistent with the values used here in the solver previously? To be consistent, it would be something like
```suggestion
        x_dx_2_xp = 2 * _dx_2 * x[i] / (2 *x[i] + (1.0/_dx) )
        x_dx_2_xm = 2 * _dx_2 * x[i] / (2 *x[i] - (1.0/_dx) )
        A[ind, ind] = - ((x_dx_2_xm + x_dx_2_xp) + 2 *_dz_2)  # j, l
        A[ind, ind + nz] = x_dx_2_xp  # j, l-1
        A[ind, ind - nz] = x_dx_2_xm  # j, l+1
```

(Possible I've made an algebra mistake, of course)

Reply:
I've inverted a few of the divisions at the beginning so this could well have been me

Reply:
Yes, the divisions switch did break stuff a little (but amazingly not any tests...)
@alexanderianblair is right though, this is a change in scheme. Honestly I was hoping the maths would pan out the same, but there are subtle differences. we discussed it and decided to stick with this one, because it's at least a more faithful implementation of the Jeon paper.

Reply:
Yup - I'm happy with the new scheme - it just appears to be a different second order discretisation, but still solves the same problem.